### ISSUE TITLE
feat: COPY using new 

### ISSUE THREAD

**Issue by BohuTANG** [datafuselabs/databend#8635]
**Summary**

COPY plan to pipeline main logic is in:
https://github.com/datafuselabs/databend/blob/a8bcc20f78281bd9e45fd9cfa58e7c6d9d697002/src/query/service/src/interpreters/interpreter_copy_v2.rs#L296-L351

To make `COPY` more simple and distributed, the flow likes: cc @BohuTANG 
1. Move the get the files to `StageTable:: read_partitions`:
https://github.com/datafuselabs/databend/blob/a8bcc20f78281bd9e45fd9cfa58e7c6d9d697002/src/query/storages/stage/src/stage_table.rs#L92-L113

2. Change the CopyPlan to NewPlanner, Using PhysicalPlanBuilder? cc @leiysky 
```
       let builder = PhysicalPlanBuilder::new(metadata, self.ctx.clone());
        let plan = builder.build(&s_expr).await?;

        ... distributed or not
        let pipeline_builder = PipelineBuilder::create(self.ctx.clone());
        let build_res = pipeline_builder.finalize(&plan)?;
```

**Comment by Xuanwo** [comment]
Can we return a stream of `Partitions` in `read_partitions`?

**Comment by BohuTANG** [comment]
> Can we return a stream of `Partitions` in `read_partitions`?

Good idea, we can read partitions chunk by chunk and do COPY in the interpreter layer.

**Comment by BohuTANG** [comment]
I think we don't need streaming, we only need a new API for list:List(limit)，each copy only run limit files copy and purge.

**Comment by Xuanwo** [comment]
> we only need a new API for list:List(limit)

That's easy. We can stone the `ObjectStreamer` returned by OpenDAL and only take `limit` entries.

**Comment by Xuanwo** [comment]
> we only need a new API for list_files(limit)，each copy only run limit files copy and purge.

At users side, we will have `COPY FROM xxx INTO yyy LIMIT 100`?

**Comment by BohuTANG** [comment]
> At users side, we will have `COPY FROM xxx INTO yyy LIMIT 100`?

I think we don't need the user to know the underlying, because COPY will return the Status to user, tell them what's the files copies and each files statistics.

**Comment by Xuanwo** [comment]
> I think we don't need the user to know the underlying, because COPY will return the Status to user, tell them what's the files copies and each files statistics.

That's a significant change to the semantics of `COPY`. After this change, COPY will not (make sure) copy all files specified by pattern or files.

**Comment by Xuanwo** [comment]
Adding `LIMIT 1000` seems a good idea to me now. Ignoring already copied file support allows users to keep running the same query until we return no file copied in status.

**Comment by BohuTANG** [comment]
It would be better don't let the user do it in their SQL, because this is a problem of databend not the user :)
